plugins/cache-ehcache/src/main/java/org/opensearch/cache/store/disk/EhcacheDiskCache.java

```diff
@@ -136,34 +140,39 @@ private EhcacheDiskCache(Builder<K, V> builder) {
             this.threadPoolAlias = builder.threadPoolAlias;
         }
         this.settings = Objects.requireNonNull(builder.getSettings(), "Settings objects shouldn't be null");
+        this.keySerializer = Objects.requireNonNull(builder.keySerializer, "Key serializer shouldn't be null");
+        this.valueSerializer = Objects.requireNonNull(builder.valueSerializer, "Value serializer shouldn't be null");
         this.cacheManager = buildCacheManager();
-        Objects.requireNonNull(builder.getEventListener(), "Listener can't be null");
-        this.eventListener = builder.getEventListener();
-        this.ehCacheEventListener = new EhCacheEventListener<K, V>(builder.getEventListener());
+        this.ehCacheEventListener = new EhCacheEventListener<K, V>(
+            Objects.requireNonNull(builder.getRemovalListener(), "Removal listener can't be null"),
+            Objects.requireNonNull(builder.getWeigher(), "Weigher function can't be null"),
+            this.valueSerializer);
         this.cache = buildCache(Duration.ofMillis(expireAfterAccess.getMillis()), builder);
+        this.shardIdDimensionName = Objects.requireNonNull(builder.shardIdDimensionName, "Dimension name can't be null");
```

Comment:
This is not right. We shouldn't be limiting this to a single dimension that too hardcoding it to be shardId specific.